src/Store/TitleStoreConnector.re

```diff
@@ -8,58 +8,81 @@ module Core = Oni_Core;
 module Model = Oni_Model;
 
 module Actions = Model.Actions;
+module Option = Core.Utility.Option;
+module Path = Core.Utility.Path;
 
-let getDirectory = (fp: string): option(string) => {
-  let dirs =
-    Filename.dirname(fp) |> String.split_on_char(Filename.dir_sep.[0]);
-
-  List.length(dirs) - 1 |> List.nth_opt(dirs);
-};
+let withTag = (tag: string, value: option(string)) =>
+  Option.map(v => (tag, v), value);
 
 let getTemplateVariables: Model.State.t => Core.StringMap.t(string) =
   state => {
-    let initialValues = [("appName", "Onivim 2")];
-
-    let initialValues =
-      switch (Model.Selectors.getActiveBuffer(state)) {
-      | None => initialValues
-      | Some(buf) =>
-        let fp = Core.Buffer.getFilePath(buf);
-        let ret =
-          switch (fp) {
-          | None => initialValues
-          | Some(fp) =>
-            let activeEditorShort = Filename.basename(fp);
-            let parentDir = getDirectory(fp);
-
-            let initialValues = [
-              ("activeEditorShort", activeEditorShort),
-              ("activeEditorLong", fp),
-              ...initialValues,
-            ];
-
-            switch (parentDir) {
-            | None => initialValues
-            | Some(dir) => [("activeFolderShort", dir), ...initialValues]
-            };
-          };
-        switch (Core.Buffer.isModified(buf)) {
-        | false => ret
-        | true => [("dirty", "*"), ...ret]
-        };
-      };
+    let buffer = Model.Selectors.getActiveBuffer(state);
+    let filePath = Option.bind(Core.Buffer.getFilePath, buffer);
+
+    let appName = Option.some("Onivim 2") |> withTag("appName");
+
+    let dirty =
+      Option.map(Core.Buffer.isModified, buffer)
+      |> (
```

Comment:
I like the way you refactored out the `withTag` method and pipe into `fun` here - nice! 👍